index/scorch/introducer.go

```diff
@@ -323,11 +323,15 @@ func (s *Scorch) introduceMerge(nextMerge *segmentMerge) {
 
 	defer func() { _ = root.DecRef() }()
 
+	creator := "introduceMerge"
+	if nextMerge.forceCompact {
+		creator = "introduceForceMerge"
+	}
```

Comment:
looking this more, perhaps it feels better to add creator as param to introduceMerge() rather than forceCompact flag?

Reply:
indeed... thats much cleaner :)